### PR TITLE
Backup: Storage UI - i2 - Improving storage warning details

### PR DESCRIPTION
**Pull request by Initsogar**
Related to #28085

Similar to what have been implemented in [Calypso for disable access to old backups if storage is full](https://github.com/Automattic/wp-calypso/pull/72145), we are adding a similar UI in backup plugin.

## Proposed changes:
- Add days of backups saved below the storage meter pointing to Activity Log in Jetpack Cloud.
- New detailed messages explained the current storage status in the upgrade prompt section.
- Add `Backups stopped` title when backup is full. Also:
  - Replace `See your backups in the cloud` button with `Upgrade your cloud storage`
  - Replace `Or view your most recent restore point` with `Or view your most recent backup`

### Screenshots:
| Description | Screenshot |
|---|---|
| Normal (< 65% used storage) |  |
| Warning (65% - 80% used storage) |  | 
| Critical (80%-100% used storage) |  |
| Full (100% used storage) |  |
| Full - Over limit (> 100% used storage) |  |
| Backups discarded |  |

### Other information:
- [x] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [x] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

* peaFOp-wy-p2
* p1HpG7-jQY-p2

## Does this pull request change what data or activity we track or use?

No

### Unit testing:
You can run unit testing using the following command:
* JS: `jetpack test packages/backup js`
* PHP: `jetpack test packages/backup php`

### Testing instructions:
* Spin up a Jurassic Ninja site with Jetpack Beta, using Backup branch `add/backup-storage-ui-i2` with a backup plan like Jetpack Backup (1 GB or 10 GB).
* Navigate to Jetpack -> VautlPress Backup
* Once the first backup completes, you should see the main page of the plugin, including the storage meter.
* Given it is a new site, it will be ~0GB used and no upsell prompt should be displayed.

To try different scenarios for the storage meter, you could try to upload big files to your site or dispatch the events using [Redux DevTools (chrome extension)](https://chrome.google.com/webstore/detail/redux-devtools/lmhkpmbekcpmknklioeibfkpmmfibljd?hl=en). After installing it, follow these steps:
* Open the extension in Chrome
* In the dropdown in the top right, select `jetpack-backup-plugin`
* In the bottom, click on `>_` option to show the dispatcher.
image
* It should look like this:

* Then in the dispatcher editor, you can set the used storage like this (the size is represented in bytes):
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* So if you have a site with 10 GB, you can use the following sizes to test the different scenarios:
  * Normal (< 65%): 1073741824 (1GB)
  * Warning (65% - 80%): 7516192768 (7GB)
  * Critical (80%-100%): 9663676416 (9GB)
  * Full (> 100%): 10737418240 (10 GB)
 
### Test data
* Normal usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 1073741824,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Warning usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 7516192768,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Critical usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 9663676416,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 30,
        daysOfBackupsSaved: 13
    }
}
```

* Full usage level
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 7,
        daysOfBackupsSaved: 7
    }
}
```

* Backups discarded state
```javascript
{
    type: 'SITE_BACKUP_SIZE_GET_SUCCESS',
    payload: {
        size: 10737418240,
        minDaysOfBackupsAllowed: 7,
        daysOfBackupsAllowed: 7,
        daysOfBackupsSaved: 13
    }
}
```

### CODE REVIEW

**Review by keoshi — DISMISSED** [review]
This is absolutely perfect, @Initsogar ! Thanks so much for the super detailed instructions of how to test this thoroughly — appreciate you.

The only thing I noticed that could be improved a tiny bit is the text under the storage bar on mobile. It's a bit cramped on smaller screens so breaking the second sentence to a new line would be ideal; like so:

Before | After
--|--
image | image

Here's the CSS I used for that in case it's helpful:

```
flex-direction: column;
gap: 4px;
```

**Comment by Initsogar** [comment]
> The only thing I noticed that could be improved a tiny bit is the text under the storage bar on mobile. It's a bit cramped on smaller screens so breaking the second sentence to a new line would be ideal;

@keoshi Thanks for catching that! I just updated it and it looks better now :)

image

File: projects/packages/backup/src/js/components/backup-storage-space/storage-usage-levels.ts [hunk 1]
```diff
@@ -20,12 +21,39 @@ const THRESHOLD_VALUES = Object.keys( THRESHOLDS )
...
 		return null;
 	}
 
+	if (
+		!! minDaysOfBackupsAllowed &&
+		!! daysOfBackupsAllowed &&
+		!! planRetentionDays &&
+		!! daysOfBackupsSaved
+	) {
+		// if current allowed days of backups is equal to the minimum, return storage full.
```
Comment (rustamshaikh) [inline projects/packages/backup/src/js/components/backup-storage-space/storage-usage-levels.ts]: I just realized that this comment is outdated (in calypso), maybe we can update this here too :)

File: projects/packages/backup/src/js/reducers/site-backup-size.js [hunk 2]
```diff
@@ -26,6 +29,9 @@ const siteBackupSize = ( state = initialState, action ) => {
 				isFetching: false,
 				loaded: true,
 				size: action.payload.size,
+				minDaysOfBackupsAllowed: action.payload.minDaysOfBackupsAllowed,
```
Comment (rustamshaikh) [inline projects/packages/backup/src/js/reducers/site-backup-size.js]:
```suggestion
				minDaysOfBackupsAllowed: action.payload?.minDaysOfBackupsAllowed,
```
Nitpick, but I think it is better to use the optional chaining operator `?.` in cases we access a nested property.

File: projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/index.jsx [hunk 3]
```diff
@@ -1,14 +1,38 @@
...
 	return (
 		<>
 			<div className="backup-storage-space__meta">
 				<div className="backup-storage-space__usage-text">{ usageText }</div>
-				<div className="backup-storage-space__retention">{ /* X days of backups saved */ }</div>
+				<div className="backup-storage-space__retention">
+					{ createInterpolateElement(
+						sprintf(
+							/* translators: %s: Number of days of backups saved. */
+							__( '<a>%s days of backups saved</a>', 'jetpack-backup-pkg' ),
```
Comment (rustamshaikh) [inline projects/packages/backup/src/js/components/backup-storage-space/storage-usage-details/index.jsx]: I think a singular version of this text would be good when `daysOfBackupsSaved=1`

**Comment by Initsogar** [comment]
@rustamshaikh thanks for the comments and all the effort reviewing this PR! I just pushed the changes for the suggestions you made. Take a look and let me know what do you think 😊

**Review by rustamshaikh — APPROVED** [review]
The changes look good to me. Tested with all the scenarios and following the test plan

**Review by sergeymitr — APPROVED** [review]
Tests well 👍

**Pull request merged**